src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/token/MeshAuthorizationToken.java

```diff
@@ -0,0 +1,32 @@
+package uk.nhs.digital.nhsconnect.nhais.mesh.token;
+
+import java.time.Instant;
+
+import uk.nhs.digital.nhsconnect.nhais.mesh.MeshConfig;
+import uk.nhs.digital.nhsconnect.nhais.service.TimestampService;
+
+/**
+ * MESH authorization token
+ * One time use only - each MESH API call should use new token
+ */
+public class MeshAuthorizationToken {
+
+    private final String data;
+    private final String hash;
+
+    public MeshAuthorizationToken(MeshConfig meshConfig, Instant timestamp, Nonce nonce, AuthorizationHashGenerator authorizationHashGenerator) {
+        String prefix = "NHSMESH " + meshConfig.getMailboxId();
+        String currentTimeFormatted = new TokenTimestamp(timestamp).getValue();
+        this.data = String.join(":", prefix, nonce.value, nonce.count, currentTimeFormatted);
+        this.hash = authorizationHashGenerator.computeHash(meshConfig, nonce, currentTimeFormatted);
+    }
+
+    public MeshAuthorizationToken(MeshConfig meshConfig) {
```

Comment:
MeshAuthorizationToken not used?

Reply:
As of now it isn't - I'll use it in integration test